regression_em/model.py

```diff
@@ -1,18 +1,209 @@
+        ref_probs = self.calc_probs(ref_params[0], ref_params[1], ref_feat)
+
+        return np.vectorize(self.calc_responsibility)(target_probs, ref_probs, labels)
+
+    def update_params(self, feat_mat: np.ndarray, responsibilities: np.ndarray, sample_weights) -> Tuple[np.ndarray, float]:
+        """Return fitted Logistic Regression params.
+
+        :param feat_mat: Feature matrix to be used to learn responsibility.
+        :param responsibilities: Sequence of responsibilities calculated at E-step.
+        :return: Updated M-step params.
+        """
+        if self.alpha:
+            reg = Ridge(alpha=self.alpha)
+        else:
+            reg = LinearRegression()
+        reg.fit(feat_mat, self.calc_logits(responsibilities), sample_weights)
+        return reg.coef_, reg.intercept_
+
+    def update_expo_params(self, feat_mat: np.ndarray, responsibilities: np.ndarray, sample_weights) -> Tuple[np.ndarray, float]:
+        """Return fitted Logistic Regression params.
 
-    x = sp.rand(100, 100)
-    lr = LinearRegression()
-    print(f"Prediction results: {rem.predict()}")
+        :param feat_mat: Feature matrix to be used to learn responsibility.
+        :param responsibilities: Sequence of responsibilities calculated at E-step.
+        :return: Updated M-step params.
+        """
+        if self.alpha:
+            reg = Ridge(alpha=self.alpha * 100)
+        else:
+            reg = LinearRegression()
+        reg.fit(feat_mat, self.calc_logits(responsibilities), sample_weights)
+        return reg.coef_, reg.intercept_
+
+    def calc_log_likelihood(self, left_feat: np.ndarray, right_feat: np.ndarray,
+                            labels: np.ndarray) -> np.ndarray:
+        """Return log likelihood.
+
+            # TODO: Describe formula here.
+
+        :param left_feat:
+        :param right_feat:
+        :param labels:
+        :return:
+        """
+        # Calculate predicted probabilities of outcome.
+        outcome_probs = self.calc_probs(self.left_params[0], self.left_params[1], left_feat) * \
+            self.calc_probs(self.right_params[0], self.right_params[1], right_feat)
+
+        # Calculate log likelihood with positive samples only.
+        positive_sample_probs = outcome_probs[labels]  # Get list of probs whose sample labels are True.
+        positive_sample_probs[positive_sample_probs == 0] = self.epsilon  # Clip probs to avoid log(0).
+        positive_sample_log_lh = np.sum(np.log(positive_sample_probs))
+
+        # Apply the same procedure to negative ones.
+        negative_sample_probs = outcome_probs[~labels]
+        negative_sample_probs[negative_sample_probs == 1] = 1 - self.epsilon
+        negative_sample_log_lh = np.sum(np.log(1 - negative_sample_probs))
+
+        return positive_sample_log_lh + negative_sample_log_lh
+
+    def fit(self, left_feat: np.ndarray, right_feat: np.ndarray, labels: np.ndarray) -> None:
```

Comment:
Follow scikit-learn UI. `fit(self, X, y)` .

Reply:
fit, predict_proba, predict functions are updated to follow scikit-learn UI.
Please review them.